lib/nimbase.h

```diff
@@ -220,8 +222,13 @@ __AVR__
 #    define N_FASTCALL_PTR(rettype, name) rettype (*name)
 #    define N_SAFECALL_PTR(rettype, name) rettype (*name)
 #  endif
-#  define N_LIB_EXPORT NIM_EXTERNC __attribute__((visibility("default")))
-#  define N_LIB_EXPORT_VAR  __attribute__((visibility("default")))
+#  ifdef __EMSCRIPTEN__
+#    define N_LIB_EXPORT  EMSCRIPTEN_KEEPALIVE __attribute__((visibility("default")))
+#    define N_LIB_EXPORT_VAR  EMSCRIPTEN_KEEPALIVE __attribute__((visibility("default")))
```

Comment:
Since the macro comes from the `emscripten.h` header, the header would need to included here, but I'm not sure whether that's a good idea.

Personally, I'd suggest replacing `EMSCRIPTEN_KEEPALIVE` with what the macro currently expands to:

```suggestion
#    define N_LIB_EXPORT __attribute__((used, visibility("default")))
#    define N_LIB_EXPORT_VAR __attribute__((used, visibility("default")))
```

Of course, this would stop working should emscripten exported C functions start to work differently.

Reply:
Okay. Sounds good